src/ucp/rndv/proto_rndv.inl

```diff
@@ -256,18 +256,13 @@ ucp_proto_rndv_bulk_max_payload(ucp_request_t *req,
         scaled_length = ucp_proto_multi_scaled_length(lpriv->weight_sum,
                                                       total_length);
 
-        ucs_assertv(scaled_length >= total_offset,
-                    "req=%p min_rndv_chunk_size=%zu scaled_length=%zu "
-                    "total_offset=%zu total_length=%zu weight_sum=%zu%% ",
-                    req, min_rndv_chunk_size, scaled_length, total_offset,
-                    total_length,
-                    ucp_proto_multi_scaled_length(lpriv->weight_sum, 100));
         /**
          * max_payload is later capped by remaining request length when
-         * advancing datatype iterator
+         * advancing datatype iterator, potential underflow is fine.
          */
         max_payload = ucs_max(min_rndv_chunk_size,
                               scaled_length - total_offset);
+        max_payload = ucs_min(max_payload, lpriv->max_frag);
```

Comment:
afair it is cut later in `dt_iter_next`, so the decision in prev PR was to avoid extra branch here

Reply:
Yes, we still don't check if `min_rndv_chunk_size` is bigger than the remaining length at this will be enforced in `dt_iter_next`. And `min_rndv_chunk_size` was assumed to be smaller than `lrpiv->max_frag` anyways.

But now we realize that it's actually possible that `max_payload` is a huge number due to underflow in the substraction line 264. Even if it is later cut, for bigger requests sizes, the number could still be bigger than `lpriv->max_frag`.

 Do you mean that `lpriv->max_frag` is later enforced on the obtained `max_payload`?

Reply:
if scaled_length < total_offset, line 264 would result in a negative value, how would that work given that all variables are unsigned types?

Reply:
all values are unsigned with defined underflow, so if `scaled_length` < `total_offset`, we get huge value, capped at lane `max_frag` by line 265. Later, dt code path make sure we remain within request length too.

we could add a check if explicit behavior is more desirable.

Reply:
ok. but compared to the situation before https://github.com/openucx/ucx/commit/4909af673af7f2e3dcd2f98410213374f53d9fca, we have 2 more branches now. Any way we can optimize this?

Reply:
tried to make it more explicit.

Reply:
now that I reread, I wonder if we should instead go away from using `scaled_length` values (accumulated weights) to rather use individual weights. With current patch, I think we risk forcing using `lpriv->max_frag` for the lane after which `min_rndv_chunk_size` was used. Maybe the lengths considered are anyways small so we don't really care and the PR is good enough. wdyt?

Reply:
finally fixed as discussed offline by pre-computing at multi init time.